File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        int maxcount = 0;
        int count = 0;

        for(int i = 0; i < n; i++){
            if(nums[i] == 1){
                count++;
                maxcount = max(count, maxcount);
            }
            else if(nums[i] != 1) {
                count = 0;
            }
        }
        return maxcount;
        
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        sum = 0
        for i in range(len(nums)):
            if nums[i] == 1:
                sum += 1
                count = max(sum, count)
            else:
                sum = 0

        return count    
        